src/ossl/sshkdf.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 Simo Sorce
+// See LICENSE.txt file for terms
+
+impl Derive for SSHKDFOperation {
+    fn derive(
+        &mut self,
+        key: &Object,
+        template: &[CK_ATTRIBUTE],
+        _mechanisms: &Mechanisms,
+        objfactories: &ObjectFactories,
+    ) -> KResult<Vec<Object>> {
+        if self.finalized {
+            return err_rv!(CKR_OPERATION_NOT_INITIALIZED);
+        }
+        self.finalized = true;
+
+        key.check_key_ops(CKO_SECRET_KEY, CKK_GENERIC_SECRET, CKA_DERIVE)?;
+
+        let mut dkey = objfactories.derive_key_from_template(key, template)?;
+        let keysize = match dkey.get_attr_as_ulong(CKA_VALUE_LEN) {
+            Ok(n) => n as usize,
+            Err(_) => return err_rv!(CKR_TEMPLATE_INCOMPLETE),
+        };
+        if keysize == 0 || keysize > (u32::MAX as usize) {
+            return err_rv!(CKR_KEY_SIZE_RANGE);
+        }
+
+        let params = OsslParam::with_capacity(5)
```

Comment:
```suggestion
        let params = OsslParam::with_capacity(6)
```
to include the terminating NULL record?

Reply:
ah, good point